airmozilla/popcorn/tests/test_views.py

```diff
@@ -157,6 +157,47 @@ def test_popcorn_editor(self):
         response = self.client.get(url, {'slug': event.slug})
         eq_(response.status_code, 200)
 
+    def test_save_edit(self):
+        event = Event.objects.get(title='Test event')
+        event.template.name = 'this is a vid.ly video'
+        event.template.save()
+
+        event.template_environment = {'tag': 'abc123'}
+        event.save()
+
+        url = reverse('popcorn:save_edit')
+
+        self._login()
+
+        response = self.client.post(url, {
+            'slug': 'does_not_exist',
+        })
+        eq_(response.status_code, 400)
+
+        response = self.client.post(url, {
+            'data': '{}',
+        })
+        eq_(response.status_code, 400)
+
+        response = self.client.post(url, {
+            'slug': 'does_not_exist',
+            'data': '{}',
+        })
+        eq_(response.status_code, 404)
+
+        response = self.client.post(url, {
+            'slug': event.slug,
+            'data': '{}',
```

Comment:
We should check that the string can be converted to a python dict. 
I think the `PopcornEdit.data` field is a JSONField which should hopefully barf but the error might then become an Internal Server Error rather than a 400 Bad Request.